linera-execution/src/wasm/mod.rs

```diff
@@ -30,30 +31,26 @@ use crate::{
     SessionId, UserApplication, WasmRuntime,
 };
 use async_trait::async_trait;
-use std::path::Path;
+use std::{path::Path, sync::Arc};
 use thiserror::Error;
 
 /// A user application in a compiled WebAssembly module.
 pub enum WasmApplication {
     #[cfg(feature = "wasmer")]
     Wasmer {
-        contract_engine: ::wasmer::Engine,
-        contract_module: ::wasmer::Module,
-        service_engine: ::wasmer::Engine,
-        service_module: ::wasmer::Module,
+        contract: Arc<(::wasmer::Engine, ::wasmer::Module)>,
```

Comment:
Can you document what's going on with contract engine? Do we need to file a follow-up issue? or a bug to Wasmer?

Reply:
Done. I think it's like that by design, because the issue is that it triggers a `panic` in the metering middleware: https://docs.rs/wasmer-middlewares/latest/wasmer_middlewares/metering/struct.Metering.html#panic